docs: add config suggestion and deployment tutorial (#284)
mosecorg/mosec: README.md

- [Multi-stage workflow demo](https://github.com/mosecorg/mosec/blob/main/examples/echo.py): a simple CPU demo.
- [Shared memory IPC](https://github.com/mosecorg/mosec/blob/main/examples/plasma_shm_ipc.py)
- [Customized GPU allocation](https://github.com/mosecorg/mosec/blob/main/examples/custom_env.py): deploy multiple replicas, each using different GPUs
- [Customized metrics](https://github.com/mosecorg/mosec/blob/main/examples/python_side_metrics.py)
- [Jax jitted inference](https://github.com/mosecorg/mosec/blob/main/examples/jax_single_layer.py)
- PyTorch deep learning models:
  - [sentiment analysis](https://github.com/mosecorg/mosec/blob/main/examples/distil_bert_server_pytorch.py): a NLP demo.
  - [image recognition](https://github.com/mosecorg/mosec/blob/main/examples/resnet50_server_msgpack.py): a CV demo.
  - [stable diffusion](https://github.com/mosecorg/mosec/tree/main/examples/stable_diffusion): with msgpack serilization.

## Configuration

> - Dynamic batch

Reviewer: ```suggestion
- Dynamic batching
```

  - `max_batch_size` is configured when you `append_worker` (make sure inference with the max value won't cause the out-of-memory in GPU)
  - `--wait (default=10ms)` is configured through CLI arguments (this usually should <= one batch inference duration)

>   - If enabled, it will collect a batch whether when it reaches the `max_batch_size` or the `wait` time.

Reviewer: ```suggestion
  - If enabled, it will collect a batch either when it reaches the `max_batch_size` or the `wait` time.
```

- Check the [arguments doc](https://mosecorg.github.io/mosec/argument/).

## Deployment

> - This may requires some shared memory, remember to set the `--shm-size` for docker.

Reviewer: ```suggestion
- This may require some shared memory, remember to set the `--shm-size` flag if you are using docker.
```

- This service doesn't require Gunicorn or NGINX, but you can certainly use the ingress controller. BTW, it should be the PID 1 process in the container since it controls multiple processes.
- Remember to collect the **metrics**.
  - `mosec_service_batch_size_bucket` shows the batch size distribution.

>   - `mosec_service_process_duration_second_bucket` shows the duration for each stage (exclude the IPC time).

Reviewer: ```suggestion
  - `mosec_service_process_duration_second_bucket` shows the duration for each stage (excluding the IPC time).
```
